src/main/java/uk/gov/ch/oauth/Oauth2.java

```diff
@@ -30,15 +34,15 @@
 public class Oauth2 implements IOauth {
 
     private static final Logger LOGGER = LoggerFactory.getLogger("docs.developer.ch.gov.uk");
-    final IIdentityProvider identityProvider;
+    private final IIdentityProvider identityProvider;
+    private final SessionUtils sessionUtils;
     private final Duration timeoutDuration = Duration.ofSeconds(10L);
+    private final NonceGenerator nonceGenerator = new NonceGenerator();
```

Comment:
Is it worth considering auto wiring the `NonceGenerator` to give us the testing option of known Nonces very easily?

Reply:
Possibly.
I think we may want a more general discussion on how we're expecting this library to be integrated into applications as our autowiring requires everywhere to setup config for those beans.